libs/ngrx-toolkit/src/lib/rx-mutation.ts

```diff
@@ -0,0 +1,171 @@
+ *       onError: (error) => {
+ *         console.error('Error occurred:', error);
+ *       },
+ *     }),
+ *   })),
+ * );
+ *
+ * function calcSum(a: number, b: number): Observable<number> {
+ *   return of(a + b);
+ * }
+ * ```
+ *
+ * @param options
+ * @returns
+ */
+export function rxMutation<P, R>(
+  options: RxMutationOptions<P, R>,
+): Mutation<P, R> {
+  const inputSubject = new Subject<{
+    param: P;
+    resolve: (result: MutationResult<R>) => void;
+  }>();
+  const flatteningOp = options.operator ?? concatOp;
+
+  const destroyRef = options.injector?.get(DestroyRef) ?? inject(DestroyRef);
+
+  const callCount = signal(0);
+  const errorSignal = signal<unknown>(undefined);
+  const idle = signal(true);
+
+  const status = computed<MutationStatus>(() => {
+    if (idle()) {
+      return 'idle';
+    }
+    if (callCount() > 0) {
+      return 'processing';
+    }
+    if (errorSignal()) {
+      return 'error';
+    }
+    return 'success';
+  });
+
+  const initialInnerStatus: MutationStatus = 'idle';
+  let innerStatus: MutationStatus = initialInnerStatus;
+  let lastResult: R;
+
+  inputSubject
+    .pipe(
+      flatteningOp.rxJsOperator((input) =>
+        defer(() => {
+          callCount.update((c) => c + 1);
+          idle.set(false);
+          return options.operation(input.param).pipe(
+            tap((result: R) => {
+              options.onSuccess?.(result, input.param);
+              innerStatus = 'success';
+              lastResult = result;
+            }),
+            catchError((error: unknown) => {
+              options.onError?.(error, input.param);
+              errorSignal.set(error);
+              innerStatus = 'error';
+              return EMPTY;
+            }),
+            finalize(() => {
+              callCount.update((c) => c - 1);
+
+              if (innerStatus === 'success') {
+                errorSignal.set(undefined);
+                input.resolve({
+                  status: 'success',
+                  value: lastResult,
+                });
+              } else if (innerStatus === 'error') {
+                input.resolve({
+                  status: 'error',
+                  error: errorSignal(),
+                });
+              } else {
+                input.resolve({
+                  status: 'aborted',
+                });
+              }
+
+              innerStatus = initialInnerStatus;
+            }),
+          );
+        }),
+      ),
+      takeUntilDestroyed(destroyRef),
+    )
+    .subscribe();
+
+  const mutationFn = (param: P) => {
+    return new Promise<MutationResult<R>>((resolve) => {
+      if (callCount() > 0 && flatteningOp.exhaustSemantics) {
+        resolve({
+          status: 'aborted',
+        });
+      } else {
+        inputSubject.next({
+          param,
+          resolve,
+        });
+      }
+    });
+  };
+
+  const mutation = mutationFn as Mutation<P, R>;
+  mutation.status = status;
+  mutation.callCount = callCount;
```

Comment:
Can we remove that one now? If we do it later for standalone, it would be breakign change.

Reply:
Done